src/main/java/uk/gov/ons/census/action/poller/ChunkProcessor.java

```diff
@@ -33,8 +43,31 @@ public void processChunk() {
     }
   }
 
+  @Transactional(propagation = Propagation.REQUIRES_NEW) // Start a new transaction for every chunk
+  public void processFulfilmentChunk() {
+    try (Stream<FulfilmentsToSend> fulfilments =
+        fulfilmentToSendRepository.findChunkToProcess(chunkSize)) {
+      fulfilments.forEach(
```

Comment:
one funky little performance optimisation we could try would be to do `fulfilments.parallel().forEach` which would give us some multithreading